sklearn/feature_selection/_variance_threshold.py

```diff
@@ -84,6 +84,8 @@ def fit(self, X, y=None):
             # for constant features
             compare_arr = np.array([self.variances_, peak_to_peaks])
             self.variances_ = np.nanmin(compare_arr, axis=0)
+        elif self.threshold < 0.:
+            raise ValueError("Threshold must be non-negative.")
```

Comment:
```suggestion
            raise ValueError(f"Threshold must be non-negative."
                             f" Got: {self.threshold}")
```

Reply:
Thank you for the feedback! I included a value in the error message.